### PR TITLE
added modal component

### PR DESCRIPTION
**Pull request by nikhiljugale007**
Please review my code for modal component

PREVIEW: https://vercel-ui-preview.netlify.app/component-lib/modal.html

### CODE REVIEW

File: component-lib/modal.html [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+		<meta http-equiv="X-UA-Compatible" content="IE=edge" />
+		<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+		<title>Modal | Docs</title>
+		<link rel="stylesheet" href="./all-comp.css" />
+		<link
+			href="https://maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css"
+			rel="stylesheet"
+		/>
+		<script>
+			var themeDefault = window.localStorage.getItem("data-theme");
```
Comment (anasm11) [inline component-lib/modal.html:14]: js can be put in different file

**Review by anasm11 — COMMENTED** [review]
better to merge in a different branch before main. The code looks good.

File: component-lib/modal.html [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+									style="
+										width: 680px;
+										height: 652px;
+										border: 0;
+										transform: scale(1);
+										overflow: hidden;
+									"
```
Comment (Ankur9669) [inline component-lib/modal.html]: The inline styling could be removed

File: component-lib/modal.html [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+									</button>
+								</div>
+								<div class="modal-body">
+									<p class="typo-subtext">
+										Model body goes here. The modal component provides a solid
+										foundation for creating dialogs, popovers, lightboxes, or
+										whatever else.
+									</p>
+								</div>
+								<div class="modal-footer">
```
Comment (Ankur9669) [inline component-lib/modal.html:212]: There could be more vertical spacing beetween footer and text

**Review by Ankur9669 — APPROVED** [review]
The code and UI is awesome, some changes that i would recommend are...
1. There could be more vertical spacing between footer and the text above as currently it is looking a little inconsistent
2. You could remove inline styling from all the iframes
3. In modal.css line 7 you have used **float: left**. I think you can remove that
4. In js you could use **let instead of var**

File: component-lib/modal.html (hunk 1 above)
Comment (nikhiljugale007) [reply component-lib/modal.html:14]: As of now, I have put js in the same file. When I get all my components reviewed I am going to change my folder structure that time I will put js in a separate file.

File: component-lib/modal.html (hunk 2 above)
Comment (nikhiljugale007) [reply component-lib/modal.html]: Moved inline styles to the common CSS file in new commit.

File: component-lib/modal.html (hunk 3 above)
Comment (nikhiljugale007) [reply component-lib/modal.html:212]: Added more spacing in new commit.

**Comment by nikhiljugale007** [comment]
> The code and UI is awesome, some changes that i would recommend are...
> 
> 1. There could be more vertical spacing between footer and the text above as currently it is looking a little inconsistent
> 2. You could remove inline styling from all the iframes
> 3. In modal.css line 7 you have used **float: left**. I think you can remove that
> 4. In js you could use **let instead of var**

Implemented all changes you suggested and made a new commit.  Can you review changes made.

**Pull request merged**